spec/support/hanami_commands.rb

```diff
@@ -15,6 +15,7 @@ def rackup(args = {}, &blk)
       end
 
       def server(args = {}, &blk)
+        args[:port] ||= rand(2300..9999)
```

Comment:
`require 'prime'`

```ruby
args[:port] ||= Prime.each(9999).to_a.map{|p| p if p > 2300}.compact.sample
```
could work and have a smaller chance of collision with another port being used